crypto/x509/x509_vfy.c

```diff
@@ -224,6 +231,15 @@ static int verify_chain(X509_STORE_CTX *ctx)
                                   ctx->param->flags);
     CB_FAIL_IF(err != X509_V_OK, ctx, NULL, ctx->error_depth, err);
 
+    /* In the non-DANE case (re-)check direct trust in the root of the chain */
```

Comment:
The DANE case ca be more strict than non-DANE when the matched TLSA record is PKIX-TA(0) or PKIX-EE(1) so such a check would need to also happen in that case (and not with DANE-TA(2)/DANE-EE(3)).  BUT: I think this PR is not warranted.  Test coverage should ensure that build chains correctly, and I don't see much point in doing duplicate work at runtime.

Reply:
Well, I suggest postponing the further handling of this PR including the fundamental discussion to post-3.0.
I just rebased this PR recently in order to fix merge conflicts before more such conflicts pile up.